src/components/TodoContext.tsx

```diff
@@ -0,0 +1,63 @@
+import React, {
+  Dispatch,
+  createContext,
+  useCallback,
+  useMemo,
+  useState,
+} from 'react';
+import { Todo } from '../types/Todo';
+
+export const TodosContext = createContext({
+  list: [] as Todo[],
+  setList: (() => {}) as Dispatch<React.SetStateAction<Todo[]>>,
+  errorMessage: '',
+  setErrorMessage: (() => {}) as Dispatch<React.SetStateAction<string>>,
+  tempTodo: {} as Todo | null,
+  setTempTodo: (() => {}) as Dispatch<React.SetStateAction<Todo | null>>,
+  editingTodo: {} as Todo | null,
+  setEditingTodo: (() => {}) as Dispatch<React.SetStateAction<Todo | null>>,
+  handleError: (_message: string) => undefined,
+  loadingTodosIds: [] as number[],
+  setLoadingTodosIds: (() => {}) as Dispatch<React.SetStateAction<number[]>>,
```

Comment:
Don't use `as`, you can define types for the context
https://blog.logrocket.com/how-to-use-react-context-typescript/###